src-built-in/components/linker/src/linker.tsx

```diff
@@ -0,0 +1,75 @@
+/*!
+* Copyright 2017 by ChartIQ, Inc.
+* All rights reserved.
+*/
+import * as React from "react";
+import { useEffect } from "react";
+import { Provider, useSelector, useDispatch } from 'react-redux';
+import * as ReactDOM from "react-dom";
+
+import "./css/linkerWindow.css";
+import "../../../../assets/css/font-finance.css";
+import "../../../../assets/css/finsemble.css";
+import { RootState } from '../../UIAPI/types';
+import store from '../../UIAPI/store';
+import * as linkerActions from "../../UIAPI/actions/linkerActions";
+
+declare var FSBL: any;
+declare global {
+    interface Window {
+        FSBL: any
+    }
+};
+
+const LinkerRefactored = () => {
+	const linker = useSelector((state: RootState) => state.linker);
+	const dispatch = useDispatch();
+	
+	const toggleChannel = (linkerIndex: number) => {
+		dispatch(linkerActions.toggleChannel(linkerIndex));
+	};
+
+    useEffect(() => {
+		dispatch(linkerActions.init());
+        return () => {
+            dispatch(linkerActions.cleanUp());
+        }
+	}, []);
+
+	const allChannels = Object.values(linker.channels);
+    const channelsElements = allChannels.map(({color, name, active, id}) => {
+        const groupClass = `linkerGroup ${color}`;
+        const style = {
+            backgroundColor: color,
+            border: `1px solid ${color}`
```

Comment:
what if color is undefined?

Reply:
If the user configures any linker channel, they need to specify its name and color in the config. The original code is not validating this color value so I didn't do anything here. If you think we should specify a default color I can consult Dan Nicolai on which color to use.